draft-ietf-quic-recovery.md

```diff
@@ -1407,15 +1409,23 @@ Invoked when an ACK frame with an ECN section is received from the peer.
 Invoked from DetectLostPackets when packets are deemed lost.
 
 ~~~
-   InPersistentCongestion(largest_lost_packet):
+   InPersistentCongestion(lost_packets):
      pto = smoothed_rtt + max(4 * rttvar, kGranularity) +
        max_ack_delay
      congestion_period = pto * kPersistentCongestionThreshold
+
+     earliest_lost_packet = lost_packets.first()
+     largest_lost_packet = lost_packets.last()
+
      // Determine if all packets in the time period before the
      // newest lost packet, including the edges, are marked
      // lost
-     return AreAllPacketsLost(largest_lost_packet,
-                              congestion_period)
+     all_packets_lost =
+       lost_packets.length() == (lost_packets.last().packet_number -
+                                 lost_packets.first().packet_number + 1)
```

Comment:
This assumes you don't send with gaps. I personally don't think we need complete psuedocode here. I think text should be enough. I feel like this logic could be very dependent on the implementation.

Reply:
I'm tending to agree with @nibanks, I'm not sure there's anything we can add here which will meaningfully clarify the pseudocode without making it quite implementation dependent.

Reply:
Yes, this gets too detailed very fast, and I don't think more code is the answer. @ghedo: Is there more we can add to the comment above to help implementers?

Reply:
IMO the text or the comment should at the very least include the points that were raised in this disucssion, otherwise we'll just keep revisiting this over and over, and discussions on GitHub really won't help implementers once the spec is final. Also, I think the slight restructuring of the pseudo-code, where `InPersistentCongestion()` takes the whole list of lost packets is also more helpful to clarify what the information needed to make the decision is.

I can look into making another PR to do that, while keeping `AreAllPacketsLost()` undefined, if no one else wants to try.

Reply:
I think that's the best way forward, if you're willing to write a PR for that.

Reply:
Yeah, I can do that, just need to find some time :) Will close this PR in the meantime.